Fastly/fastly_waf/connector_fastly_waf.py

```diff
@@ -0,0 +1,168 @@
+import datetime
+import time
+from functools import cached_property
+from typing import Generator
+
+import orjson
+from dateutil.parser import isoparse
+from pydantic import Field
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.storage import PersistentJSON
+
+from .client import ApiClient
+from .metrics import EVENTS_LAG, FORWARD_EVENTS_DURATION, INCOMING_MESSAGES, OUTCOMING_EVENTS
+
+
+class FastlyWAFConnectorConfiguration(DefaultConnectorConfiguration):
+    email: str = Field(..., description="User's email")
+    token: str = Field(..., description="API token")
+    corp: str = Field(..., description="Corporation name", pattern=r"^[0-9a-z_.-]+$")
+    site: str = Field(..., description="Site name", pattern=r"^[0-9a-z_.-]+$")
+
+    frequency: int = 60
+    chunk_size: int = 1000
+
+
+class FastlyWAFConnector(Connector):
+    base_uri = "https://dashboard.signalsciences.net"
+    configuration: FastlyWAFConnectorConfiguration
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.from_datetime = self.most_recent_date_seen
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(email=self.configuration.email, token=self.configuration.token)
+
+    @property
+    def most_recent_date_seen(self) -> datetime.datetime:
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        with self.context as cache:
+            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+
+        # if undefined, retrieve events from the last minute
+        if most_recent_date_seen_str is None:
+            return now - datetime.timedelta(minutes=1)
+
+        # parse the most recent date seen
+        most_recent_date_seen = isoparse(most_recent_date_seen_str)
+
+        # We don't retrieve messages older than one week
+        one_week_ago = now - datetime.timedelta(days=7)
+        if most_recent_date_seen < one_week_ago:
+            most_recent_date_seen = one_week_ago
```

Comment:
The documentation of the API doesn't define a lower limit. I suggest increasing the catch-back period to 30 days.
```suggestion
        one_month_ago = now - datetime.timedelta(days=30)
        if most_recent_date_seen < one_month_ago:
            most_recent_date_seen = one_month_ago
```